File: c/problem_62.c
#include "problem_62.h"
#include <stdio.h>
#include <assert.h>

long long count_paths(int n, int m) {
    long long dp[16][16] = {{0}};
    for (int i=0;i<n;i++) dp[i][0]=1;
    for (int j=0;j<m;j++) dp[0][j]=1;
    for (int i=1;i<n;i++) for(int j=1;j<m;j++) dp[i][j]=dp[i-1][j]+dp[i][j-1];
    return dp[n-1][m-1];
}

void problem62(void) {
    printf("Problem 62:\n");
    assert(count_paths(2,2)==2);
    assert(count_paths(5,5)==70);
    assert(count_paths(1,1)==1);
    printf("All Tests Passed\n");
}

File: lua/problem_62.lua
local M = {}

function M.count_paths(n, m)
    local dp = {}
    for i = 1, n do
        dp[i] = {}
        for j = 1, m do dp[i][j] = 1 end
    end
    for i = 2, n do
        for j = 2, m do
            dp[i][j] = dp[i-1][j] + dp[i][j-1]
        end
    end
    return dp[n][m]
end

function M.problem62()
    print("Problem 62:")
    assert(M.count_paths(2, 2) == 2)
    assert(M.count_paths(5, 5) == 70)
    assert(M.count_paths(1, 1) == 1)
    print("All Tests Passed")
end

return M
